✅ Status-bar protractor tests

dominikkaminski opened the pull request:
**Changes:**

1. Upgrade protractor version
2. Use Jasmine 2.x
3. Add Protractor parameter (aureliaUrl, defaultTimeoutMS)
4. Add Integration test for status-bar

## How can others test the changes?

```shell
npm run integration-test-init
npm run integration-test
```

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

alexanderkasten commented on test/e2e/src/statusBar.spec.ts:
```diff
@@ -0,0 +1,44 @@
+import * as path from 'path';
+import { browser, by, element, ElementFinder, protractor, ProtractorExpectedConditions } from 'protractor';
```
please remove the whitespaces between the brackets

cmg-dev replied:
@alexanderkasten Wouldn't it be more useful to show and implement with @dominikkaminski the tslint setup ;)

Rhetorical question: Do it @alexanderkasten

cmg-dev commented on test/e2e/src/statusBar.spec.ts:
```diff
@@ -0,0 +1,44 @@
+  });
+
+  it('should contain settings button', () => {
+    const settingsButton: ElementFinder = statusBar.element(by.className('status-bar__element')).element(by.tagName('a'));
```
This will be fragile, as it binds the assert very hard to the HTML structure.

Could we use another mechanism?

Maybe IDs?

Vyperus replied:
We already discussed this and @dominikkaminski is already working on an ID solution for this.
This means that this will be refactored into using IDs in the next PR.

dominikkaminski commented:
@cmg-dev @alexanderkasten  @Vyperus Please recheck this PR due to code changes.

@cmg-dev I added IDs in UI

@alexanderkasten I removed whitespaces
